lib/ansible/modules/notification/office_365_connector_card.py

```diff
@@ -0,0 +1,291 @@
+OFFICE_365_CARD_TYPE = "MessageCard"
+
+
+def build_actions(actions):
+    action_items = []
+
+    for action in actions:
+        action_item = snake_dict_to_camel_dict(action)
+        action_items.append(action_item)
+
+    return action_items
+
+
+def build_sections(sections):
+    sections_created = []
+
+    for section in sections:
+        sections_created.append(build_section(section))
+
+    return sections_created
+
+
+def build_section(section):
+    section_payload = dict()
+
+    if 'title' in section:
+        section_payload['title'] = section['title']
+
+    if 'start_group' in section:
+        section_payload['startGroup'] = section['start_group']
+
+    if 'activity_image' in section:
+        section_payload['activityImage'] = section['activity_image']
+
+    if 'activity_title' in section:
+        section_payload['activityTitle'] = section['activity_title']
+
+    if 'activity_subtitle' in section:
+        section_payload['activitySubtitle'] = section['activity_subtitle']
+
+    if 'activity_text' in section:
+        section_payload['activityText'] = section['activity_text']
+
+    if 'hero_image' in section:
+        section_payload['heroImage'] = section['hero_image']
+
+    if 'text' in section:
+        section_payload['text'] = section['text']
+
+    if 'facts' in section:
+        section_payload['facts'] = section['facts']
+
+    if 'images' in section:
+        section_payload['images'] = section['images']
+
+    if 'actions' in section:
+        section_payload['potentialAction'] = build_actions(section['actions'])
+
+    return section_payload
+
+
+def build_payload_for_connector_card(module, summary, color, title, text, actions, sections):
+    payload = dict()
+    payload['@context'] = OFFICE_365_CARD_CONTEXT
+    payload['@type'] = OFFICE_365_CARD_TYPE
+
+    if summary is not None:
+        payload['summary'] = summary
+
+    if color is not None:
+        payload['themeColor'] = color
+
+    if title is not None:
+        payload['title'] = title
+
+    if text is not None:
+        payload['text'] = text
+
+    if actions:
+        payload['potentialAction'] = build_actions(actions)
+
+    if sections:
+        payload['sections'] = build_sections(sections)
+
+    payload = module.jsonify(payload)
+    return payload
+
+
+def do_notify_connector_card_webhook(module, webhook, payload):
+    headers = {
+        'Content-Type': 'application/json'
+    }
+
+    response, info = fetch_url(
+        module=module,
+        url=webhook,
+        headers=headers,
+        method='POST',
+        data=payload
+    )
+
+    if info['status'] == 200:
+        module.exit_json(changed=True)
+    else:
+        module.fail_json(
+            msg="failed to send %s as a connector card to Incoming Webhook: %s"
+                % (payload, info['msg'])
+        )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook=dict(required=True, no_log=True),
```

Comment:
@dagwieers @s-hertel The default **is** str.  It changed (from what is now type=raw) a while ago, maybe 2.0 or 2.1.  One of the major drivers for the change was that while yaml has types, k=v parsing does not.  things like ports and timeouts were then getting to the modules as ints if they were specificed via yaml or strings if they were specified via k=v parsing.

There are a few parameters in a few modules (for instance, the infamous file mode parameter) which use type=raw so that the module can accept different types of values for the parameter.  We try not to use that in modules if we can get away with it but there are a few places where it's needed.